### PR TITLE
Resize Crop Mirror

### PR DESCRIPTION
**Pull request by r-abishek**
- Adds fused Tensor Resize Crop Mirror on HIP and HOST
- Adds relevant unit and perf tests

### CODE REVIEW

File: src/modules/cpu/kernel/resize_crop_mirror.hpp [hunk 1]
```diff
@@ -0,0 +1,1012 @@
...
+                pWeightParams[0] = _mm256_set1_ps(weightParams[0]);
+                pWeightParams[1]  = _mm256_set1_ps(weightParams[1]);
+                pDstLoc = pDstLocInit;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+                    __m256 pSrc[12], pDst[3];
+                    __m256i pxSrcLoc;
+                    computeFnSrcLocAvx(pDstLoc, pWRatio, srcLocationColumnArray, &pWeightParams[2], pxSrcLoc, pWOffset, true);   // Compute the src col location correspoding to the dst col location
```
Comment (rrawther) [inline src/modules/cpu/kernel/resize_crop_mirror.hpp:88]: Is pDstLoc changing with every call of the helper fn? Else use pDstLocInit instead

File: src/modules/cpu/kernel/resize_crop_mirror.hpp [hunk 2]
```diff
@@ -0,0 +1,1012 @@
...
+        Rpp32s vectorIncrementPkd = 24;
+
+        Rpp8u *srcPtrChannel, *dstPtrChannel, *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        Rpp32u alignedLength = dstImgSize[batchCount].width & ~7;   // Align dst width to process 8 dst pixels per iteration
+        __m256 pWRatio = _mm256_set1_ps(wRatio);
```
Comment (rrawther) [inline src/modules/cpu/kernel/resize_crop_mirror.hpp:43]: usually pXXX is for denoting pointers. Suggest regular names instead

**Review by rrawther — COMMENTED** [review]
@asalmanp to review HIP kernel

File: src/modules/cpu/kernel/resize_crop_mirror.hpp (hunk 1 above)
Comment (r-abishek) [reply src/modules/cpu/kernel/resize_crop_mirror.hpp:88]: Yeah, pDstLoc changes inside the helper, so an initial value has been set as pDstLocInit, and pDstLoc is used inside - https://github.com/r-abishek/rpp/blob/ar/opt_resize_crop_mirror/src/include/cpu/rpp_cpu_common.hpp#L4470

File: src/modules/cpu/kernel/resize_crop_mirror.hpp (hunk 2 above)
Comment (r-abishek) [reply src/modules/cpu/kernel/resize_crop_mirror.hpp:43]: I have a convention renaming PR in mind at some point for the HOST side (like we did a few months ago for the hip). Shall probably do a global renaming. 
Currently, we've got the same "pVariable" for __m256 and "pxVariable" for __m256i that we've been using for a long time.

**Comment by kiritigowda** [comment]
@asalmanp & @paveltc let me know when this is ready to be merged.

**Comment by paveltc** [comment]
@kiritigowda @r-abishek @asalmanp @rrawther This PR passes the relevant unit tests.

**Pull request merged**